pyproject.toml

```
@@ -1,15 +1,82 @@
 [build-system]
-build-backend = "setuptools.build_meta"
+build-backend = "mesonpy"
 requires = [
-    "setuptools>=42",
-    "setuptools_scm[toml]>=7.0.0",
+    "meson-python",
     "pybind11>=2.8.0",
```

Would suggest bumping `pybind11` and adding `numpy` (particularly 2) here. These are needed to make sure binaries built are NumPy 2 compatible. More details in issue ( https://github.com/pyamg/pyamg/issues/415 )

```suggestion
    "pybind11>=2.12.0,<3.0.0a0",
    "numpy>=2.0.0,<3.0.0a0",
```

The upper bounds seems like a good idea to avoid surprises, but that can be up to personal preference